### DISCUSSION TITLE
Question: Hash Routing

### DISCUSSION THREAD

**Discussion by Alex-Wenner-FHR** [h2oai/wave#843] [Q&A]
If I initialize my client, app, and users on my app.py when I re route do specific initializations need to be done again?

**Comment by Alex-Wenner-FHR** [comment]
This is a better place for this. Thank you for moving accordingly!

**Comment by lo5** [comment]
@AlexWennerDT - in general, [discussions](https://github.com/h2oai/wave/discussions) are a better place for questions :)

No, you shouldn't have to initialize any of those again. Is there anything specific that's not working?

**Reply by Alex-Wenner-FHR** [reply]
Agreed! Apologies for not utilizing this more. Will do that going forward. But no I noticed some strange behavior but with that answer, I will re-evaluate the way I am initializing those and continue this thread if any oddity occurs! Thanks!

**Reply by lo5** [reply]
No worries.

The pattern we follow in the examples is:
```py
if not q.app.initialized:
    # do something
    q.app.initialized = True
```
(replace `app` with `user` or `client` - same thing).

**Comment by Alex-Wenner-FHR** [comment]
Hi! I have previously a one page app in one .py file. All of the rendering functions, init functions, on functions and my serve function. I have now broken it up to where it is still a one page app, however I have separated the serve function into a new file. I am experiencing unexpected behavior with my client and app initializations. If I have two browsers open, I see two client initializations occur, and one ends up hitting an httpx.ConnectTimeout. I am initializing them the same as I was previously, the only difference is the initializations are separated from my serve function. I am puzzled as to why it is behaving this way. Any insight would be super helpful!

**Reply by Alex-Wenner-FHR** [reply]
I moved my initializations into the same file as the serve function and all is working as expected. Odd that this is the case. Not persistent but any thoughts on why this may be?

**Reply by lo5** [reply]
Hard to say without seeing example code. Wave doesn't care about how the code is organized, so refactoring code into multiple files shouldn't matter.
